consensus/src/payload_manager/quorum_store_payload_manager.rs

```diff
@@ -489,6 +531,36 @@ impl TPayloadManager for QuorumStorePayloadManager {
     }
 }
 
+/// Extracts and returns all inline transactions from the payload in the given block
+fn get_inline_transactions(block: &Block) -> Vec<SignedTransaction> {
+    // If the block has no payload, return an empty vector
+    let Some(payload) = block.payload() else {
+        return vec![];
+    };
+
+    // Fetch the inline transactions from the payload
+    match payload {
+        Payload::QuorumStoreInlineHybrid(inline_batches, ..) => {
+            // Flatten the inline batches and return the transactions
+            inline_batches
+                .iter()
+                .flat_map(|(_batch_info, txns)| txns.clone())
+                .collect()
+        },
+        Payload::QuorumStoreInlineHybridV2(inline_batches, ..) => {
+            // Flatten the inline batches and return the transactions
+            inline_batches
+                .iter()
+                .flat_map(|(_batch_info, txns)| txns.clone())
+                .collect()
+        },
+        Payload::OptQuorumStore(opt_qs_payload) => opt_qs_payload.inline_batches().transactions(),
```

Comment:
@ibalajiarun to double check if this is sufficient, if QS filters out batches I assume optqs would just fail to get_transactions and timeout?